cli/cmd/upgrade.go

```diff
@@ -438,6 +438,18 @@ type issuerData struct {
 	exp time.Time
 }
 
+func verifyCreds(creds *tls.Cred, trustAnchors, dns string) error {
```

Comment:
I think there is an opportunity to reuse your new code from `pkg/issuercerts` in `readIssuer()` and `fetchIssuer()`, while getting rid of the `issuerData` struct and `verifuCreds()` function. I recommend we do the clean up in a separate PR.